app/src/main/java/com/kickstarter/viewmodels/projectpage/LatePledgeCheckoutViewModel.kt

```diff
@@ -0,0 +1,162 @@
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.kickstarter.libs.Environment
+import com.kickstarter.libs.utils.extensions.isNotNull
+import com.kickstarter.models.CreatePaymentIntentInput
+import com.kickstarter.models.Project
+import com.kickstarter.models.StoredCard
+import com.stripe.android.Stripe
+import com.stripe.android.confirmPaymentIntent
+import com.stripe.android.model.ConfirmPaymentIntentParams
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.rx2.asFlow
+
+data class LatePledgeCheckoutUIState(
+    val storeCards: List<StoredCard> = listOf(),
+    val userEmail: String = ""
+)
+
+class LatePledgeCheckoutViewModel(val environment: Environment) : ViewModel() {
+
+    private val apolloClient = requireNotNull(environment.apolloClientV2())
+
+    private var storedCards: List<StoredCard> = listOf()
+    private var userEmail: String = ""
+    private var checkoutId: String? = null
+
+    private var stripe: Stripe = requireNotNull(environment.stripe())
+
+    private var mutableLatePledgeCheckoutUIState = MutableStateFlow(LatePledgeCheckoutUIState())
+    val latePledgeCheckoutUIState: StateFlow<LatePledgeCheckoutUIState>
+        get() = mutableLatePledgeCheckoutUIState
+            .asStateFlow()
+            .stateIn(
+                scope = viewModelScope,
+                started = SharingStarted.WhileSubscribed(),
+                initialValue = LatePledgeCheckoutUIState()
+            )
+
+    init {
+        viewModelScope.launch {
+            environment.currentUserV2()?.observable()?.asFlow()?.map {
+                if (it.isPresent()) {
+                    apolloClient.userPrivacy().asFlow().map { userPrivacy ->
+                        userEmail = userPrivacy.email
+                        mutableLatePledgeCheckoutUIState.emit(
+                            LatePledgeCheckoutUIState(
+                                storeCards = storedCards,
+                                userEmail = userEmail,
+                            )
+                        )
+                    }.catch {
+                        // Some error
+                    }.collect()
+
+                    apolloClient.getStoredCards().asFlow().map { cards ->
+                        storedCards = cards
+                        mutableLatePledgeCheckoutUIState.emit(
+                            LatePledgeCheckoutUIState(
+                                storeCards = storedCards,
+                                userEmail = userEmail,
+                            )
+                        )
+                    }.catch {
+                        // Some error
```

Comment:
do we have an error path for stored cards not fetching?

Reply:
not really, i think we would just throw out a generic error, but ill double check to see what the current flow does

Reply:
on PledgeFragment we do the generic error, red snackbar with "something went wrong". Same pattern could be use throughout the late pledge chedkout flow. And can be a separated ticket :)

Reply:
I'm cool with it being a separate ticket, just something that im currently trying to address in the QA script and was curious how we were handling it

Reply:
[MBL-1306](https://kickstarter.atlassian.net/browse/MBL-1306) ticket here